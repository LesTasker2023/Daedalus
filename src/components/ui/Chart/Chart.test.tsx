import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';

describe('Chart', () => {
  it('renders without crashing', () => {
    render(<div data-testid="chart-test">Chart Test</div>);
    expect(screen.getByTestId('chart-test')).toBeInTheDocument();
  });

  describe('Snapshots', () => {
    it('matches basic snapshot', () => {
      const { container } = render(<div data-testid="chart-container">Chart Component Test</div>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('renders test content correctly', () => {
    render(<div data-testid="content">Chart Content</div>);
    expect(screen.getByTestId('content')).toHaveTextContent('Chart Content');
  });
});
