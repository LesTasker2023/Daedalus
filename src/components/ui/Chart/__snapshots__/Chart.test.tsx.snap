// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chart > Snapshots > matches basic snapshot 1`] = `
<div
  data-testid="chart-container"
>
  Chart Component Test
</div>
`;
